### PR TITLE
add all integer and bytes types

### PR DESCRIPTION
**Pull request by charles-cooper**
### What I did
resolves https://github.com/vyperlang/vyper/issues/2559

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: vyper/semantics/types/value/numeric.py [hunk 1]
```diff
@@ -126,23 +128,43 @@ def from_literal(cls, node: vy_ast.Constant) -> BaseTypeDefinition:
...
-
-class Uint8Definition(UnsignedIntegerAbstractType, _UnsignedIntegerDefinition):
-    _bits = 8
-    _invalid_op = vy_ast.USub
-
-
-class Uint256Definition(UnsignedIntegerAbstractType, _UnsignedIntegerDefinition):
-    _bits = 256
-    _invalid_op = vy_ast.USub
+for i in range(32):
```
Comment (fubuloubu) [inline vyper/semantics/types/value/numeric.py:131]: A little dirty, but I like it

**Review by fubuloubu — COMMENTED** [review]
The addition of so many type in gore's is a bit of an issue for me, but can be refactored later

File: vyper/codegen/types/types.py [hunk 2]
```diff
@@ -23,11 +23,11 @@
 from vyper.utils import ceil32
 
 # Available base types
-UNSIGNED_INTEGER_TYPES = {"uint8", "uint256"}
-SIGNED_INTEGER_TYPES = {"int128", "int256"}
+UNSIGNED_INTEGER_TYPES = {f"uint{8*(i+1)}" for i in range(32)}
```
[the remark this answers is not in the document]
Comment (charles-cooper) [reply vyper/codegen/types/types.py:26]: yep, 112 is divisible by 8

I don't think we are ever going to have int_<bits> where <bits> is not divisible by 8 as it's prohibited by the ABI.

[the remark this answers is not in the document]
Comment (fubuloubu) [reply vyper/codegen/types/types.py:26]: Agree!

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**